include/boost/geometry/io/wkt/write.hpp

```diff
@@ -113,13 +113,14 @@ struct double_closing_parenthesis
 /*!
 \brief Stream points as \ref WKT
 */
-template <typename Point, typename Policy>
+template <typename Point, typename Policy1, typename Policy2>
 struct wkt_point
 {
     template <typename Char, typename Traits>
     static inline void apply(std::basic_ostream<Char, Traits>& os, Point const& p, bool)
     {
-        os << Policy::apply() << "(";
+        if(dimension<Point>::type::value == 3) os << Policy2::apply() << "(";
+        else os << Policy1::apply() << "(";
```

Comment:
Question: do we always want a `Z` in our 3D point? Until now we just did without and it works without.
If we suddenly add a `Z` here it might be incompatible for users (if any) who use this functionality. So it's a breaking change.

Apart from this it should be a compile time condition i/o a runtime condition. So we should dispatch based on dimension. But I'm not sure about that either, because if it is optional (as it probably should be), that might be inconvenient.

Reply:
As far as I can see the specification indeed mentions a `Z` for three dimensional points. It's unclear if a three dimensional point without a `Z` is acceptable or not.
But what do we do with four dimensional points? Or even more? We've not much support for them of course, but we can calculate distances for them...

In my personal opinion, for a three dimensional point, a list with three coordinates is already enough. In reading we can read or not read the `Z` (that is already implemented).

Reply:
I would agree with @barendgehrels 

Another issue here is that after this change the following code 
```
boost::geometry::read_wkt("POINT M(1 2 3)", p);
std::cout << boost::geometry::wkt(p) << std::endl;
```
will output
    
```POINT Z(1 2 3)```

I think we should decide how we want to support more than 2 dimensions and probably issue https://github.com/boostorg/geometry/issues/664 is the right place for that.

Reply:
@barendgehrels points out a valid issue.

> But what do we do with four dimensional points?

From the OGC SFS point, we have:
- OGC SFS 1.1.0 restricted to 2D geometries 
- The extended WKT/ or 2.5D WKT implemented by OGR, PostGIS, etc. allow 2, 3, 4 dimension coordinates and do not use `Z` or `M` tags, see [Adam Gawne-Cain, 1999, Simple Features Revision Proposal](https://gis.stackexchange.com/a/79189/939)
- OGC SFS 1.2.x allows 2, 3 and 4 dimensions using the `Z` or `M` tags:  `POINT Z(x, y, z)`,   `POINT M(x, y, m)`,   `POINT ZM(x, y, z, m)`. I think many may consider `M` as pseudo-dimension or not a dimension but a measure value.

>  a list with three coordinates is already enough

It seems desirable indeed to keep the current WKT I/O that does not read/output the `Z` and `M` tags.

Perhaps, an acceptable solution would be an optional `flavor` parameter, or similar, with current

- `boost::geometry::read_wkt(wkt, geometry, flavor`
- `boost::geometry::wkt(geometry, flavor)`

where `flavor` is enum, something like:

```cpp
enum /*class*/ wkt_flavour
{
  ogc2d,        // OGC SFS 1.1.0
  ogc3d,        // OGC SFS 1.2.x
  ogc25d,       // Adam Gawne-Caine, 1999, Simple Features Revision Proposal
  ewkt = ogc25d // PostGIS terminology
};
```

or as an entirely separate OGC-compliant `boost/geometry/gis/io/wkt/` extension.